### ISSUE TITLE
smartos 20150917T232817Z unable to import zones from 20130503T001121Z 

### ISSUE THREAD

**Issue by kfr-** [joyent/smartos-live#500]
Upgrading from 20130503T001121Z to 20150917T232817Z

20150917T232817Z is unable to read the zones zpool that was previously mounted on 20130503T001121Z

zpool import does not return the zones pool but it does return older unused pools created under snc_134b

20130503T001121Z has the following features:
async_destroy                         (read-only compatible)
     Destroy filesystems asynchronously.
empty_bpobj                           (read-only compatible)
     Snapshots use less space.
lz4_compress                         
     LZ4 compression algorithm support.
filesystem_limits                     (read-only compatible)
     Filesystem and snapshot limits.

20150917T232817Z has the following features:
async_destroy                         (read-only compatible)
     Destroy filesystems asynchronously.
empty_bpobj                           (read-only compatible)
     Snapshots use less space.
lz4_compress                         
     LZ4 compression algorithm support.
multi_vdev_crash_dump                
     Crash dumps to multiple vdev pools.
spacemap_histogram                    (read-only compatible)
     Spacemaps maintain space histograms.
enabled_txg                           (read-only compatible)
     Record txg at which a feature is enabled
hole_birth                           
     Retain hole birth txg for more precise zfs send
extensible_dataset                   
     Enhanced dataset functionality, used by other features.
embedded_data                        
     Blocks which compress very well use even less space.
bookmarks                             (read-only compatible)
     "zfs bookmark" command
filesystem_limits                     (read-only compatible)
     Filesystem and snapshot limits.
large_blocks                         
     Support for blocks larger than 128KB.

**Comment by jclulow** [comment]
What is the pool topology? (`zpool status`)

What disks do you have in the system? (`diskinfo`)

**Comment by kfr-** [comment]
zpool status
  pool: vpool
 state: DEGRADED
status: One or more devices could not be opened.  Sufficient replicas exist for
        the pool to continue functioning in a degraded state.
action: Attach the missing device and online it using 'zpool online'.
   see: http://illumos.org/msg/ZFS-8000-2Q
  scan: none requested
config:

        NAME                       STATE     READ WRITE CKSUM
        vpool                      DEGRADED     0     0     0
          mirror-0                 DEGRADED     0     0     0
            9248482210922706188    UNAVAIL      0     0     0  was /dev/dsk/c2t5000C50032F7BC2Fd0s0
            c0t5000C50032F7D019d0  ONLINE       0     0     0

errors: No known data errors

  pool: zones
 state: ONLINE
  scan: scrub repaired 0 in 3h58m with 0 errors on Sat Oct 24 04:58:10 2015
config:

        NAME                       STATE     READ WRITE CKSUM
        zones                      ONLINE       0     0     0
          mirror-0                 ONLINE       0     0     0
            c0t5000C50057B6A83Fd0  ONLINE       0     0     0
            c0t5000C50057B6C1AFd0  ONLINE       0     0     0
        logs
          mirror-1                 ONLINE       0     0     0
            c0t5001B44A0FF534FBd0  ONLINE       0     0     0
            c0t5001B44A0FF53552d0  ONLINE       0     0     0

errors: No known data errors


TYPE    DISK      VID         PID                SIZE            REMV    SSD
SCSI    c0t5000C50032F273E2d0    ATA         ST31000524AS       931.51 GiB      no      no  
SCSI    c0t5001B44A0FF53552d0    ATA         SanDisk SD6SB1M0   59.63 GiB       no      yes 
SCSI    c0t5001B44A0FF534FBd0    ATA         SanDisk SD6SB1M0   59.63 GiB       no      yes 
SCSI    c0t5000C50032F7D019d0    ATA         ST31000524AS       931.51 GiB      no      no  
SCSI    c0t5000C50057B6C1AFd0    SEAGATE     ST1000NM0023       931.51 GiB      no      no  
SCSI    c0t5000C50057B6A83Fd0    SEAGATE     ST1000NM0023       931.51 GiB      no      no  
USB     c1t0d0    Kingston    DataTraveler SE9   7.27 GiB        yes     no  

 zpool import
no pools available to import

**Comment by kfr-** [comment]
vpool is no longer used... I should destroy it.

**Comment by melloc** [comment]
Hi @kfr- , we discovered a bug Monday ([OS-4866](https://smartos.org/bugview/OS-4866)) when upgrading some systems from 2013 platforms, and have backported a patch to 20151029. I'm currently building new ISO/USB images which will be finished in ~3 hours, but the platform build is done and has the patch. I don't know what the status of your upgrade is, but you'll want to make sure you redownload an image w/ the fix and use that when upgrading to avoid running into the same issue.

**Comment by kfr-** [comment]
@melloc OS-4866 doesn't sound like th bug i'm running into.

The zones pool isn't available when we upgrade to a newer smartos.

**Comment by melloc** [comment]
@kfr- , no, it isn't. Just wanted to warn you since you said you were moving from a 2013 platform to 20150917 which does have OS-4866.

**Comment by kfr-** [comment]
@melloc I upgraded one of our servers from 20141212 to 20151104 without issue.

So it seems that my zfs zones issue doesn't apply for 20141212 to 20151104

This weekend i'm going to upgrade a server from 20140221 to 20151104
